pkg/commands/compute/language_go.go

```diff
@@ -106,15 +106,51 @@ func (g *Go) DefaultBuildScript() bool {
 func (g *Go) Dependencies() map[string]string {
 	deps := make(map[string]string)
 
-	if f, err := os.Open("go.sum"); err == nil {
+	if f, err := os.Open("go.mod"); err == nil {
 		defer f.Close()
+		var insideRequireBlock bool
+
 		scanner := bufio.NewScanner(f)
 		for scanner.Scan() {
 			line := scanner.Text()
-			segs := strings.Split(line, " ")
-			if len(segs) >= 2 {
-				pkg := segs[0]
-				version := strings.Split(segs[1], "/")[0] // e.g. `v0.2.0/go.mod`
+			parts := strings.Fields(line)
+
+			// go.mod has two separate `require` definitions:
+			//
+			// 1.
+			// require github.com/fastly/compute-sdk-go v1.0.0
+			//
+			// 2.
+			// require (
+			//   github.com/fastly/compute-sdk-go v1.0.0
+			//   golang.org/x/mod v0.4.2 // indirect
+			// )
+			//
+			// We avoid including 'indirect' dependencies by ignoring lines with a
+			// length greater than three segments.
+
+			if len(parts) >= 2 && parts[0] == "require" && parts[1] == "(" {
+				insideRequireBlock = true
+				continue
+			}
+			if len(parts) == 1 && parts[0] == ")" {
+				insideRequireBlock = false
+				continue
+			}
```

Comment:
I won't die on this hill but I'll make one final pitch for `golang.org/x/mod/modfile`.

```go
data, err := os.ReadFile("go.mod")
if err != nil {
	return
}

f, err := modfile.ParseLax("go.mod", data, nil)
if err != nil {
	return
}

for _, req := range f.Require {
	if req.Indirect {
		continue
	}

	deps[req.Mod.Path] = req.Mod.Version
}
```

Reply:
Oh! Sorry, I think the use of `golang.org/x/mod/modfile` got lost in the shuffle as that definitely is a better solution using an existing `x` package! 

I'll update the code, thanks for triple-checking I saw this 👍🏻

Reply:
Done.